### PR TITLE
kv: remove the useless field of `snapshot`

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
`snapshot` isn't being used.

### What is changed and how it works?
Remove the useless field in kv.

### Check List 

Tests 
- No code

### CODE REVIEW

File: kv/union_store.go [hunk 1]
```diff
@@ -80,7 +79,6 @@ type unionStore struct {
 func NewUnionStore(snapshot Snapshot) UnionStore {
```
Comment (ngaut) [inline kv/union_store.go]: Can you remove the argument of the function?

Comment (zimulala) [reply kv/union_store.go]: It is used in `NewBufferStore`.

**Comment by zimulala** [comment]
PTAL @ngaut @tiancaiamao

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**